mixer/cmd/build.go

```diff
@@ -199,113 +213,163 @@ var buildFormatNewCmd = &cobra.Command{
-		if err != nil {
-			fail(err)
-		}
-		newFormat++
-		if err = b.UpdateFormatVersion(strconv.Itoa(newFormat)); err != nil {
+		// Change the format internally just for building bundles so the right
+		// /usr/share/defaults/swupd/format is inserted; preserve old format
+		oldFormat := b.State.Mix.Format
+		b.State.Mix.Format = buildFlags.newFormat
+
+		// Build bundles normally. At this point the bundles to be deleted should still
+		// be part of the mixbundles list and the groups.ini
+		if err = buildBundles(b, buildFlags.noSigning, buildFlags.clean); err != nil {
 			fail(err)
 		}
-		setWorkers(b)
 
-		fmt.Println(" Backing up full groups.ini")
-		// Back up groups.ini in case we have deprecated bundles to delete
-		if err = b.CopyFullGroupsINI(); err != nil {
+		// Wipes the <bundle>-info files and replaces them with
+		// <bundle>/ directories that are empty. When the manifest creation happens it will
+		// mark all files in that bundles as deleted.
+		f := func(filename string) error {
+			bdir := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer, filename)
+			// This bundle is deprecated, remove the info file
+			if err = os.RemoveAll(bdir + "-info"); err != nil {
+				return err
+			}
+			// Create the empty dir for update to mark all files as deleted
+			if err = os.MkdirAll(bdir, 0755); err != nil {
+				return errors.Wrapf(err, "Failed to create bundle directory: %s", bdir)
+			}
+			return nil
+		}
+		fmt.Println("Removing deprecated bundle-info files...")
+		// Remove deleted bundles and replace with empty dirs for update to mark as deleted
+		if err = b.ModifyBundles(f); err != nil {
 			fail(err)
 		}
-		// Fill this in w/Update bundle definitions
-		// if err := UpdateBudlesForFormatBump(); err != nil {...}
 
-		// Build the +20 (first build in new format) bundles
-		if err = buildBundles(b, buildFlags.noSigning, buildFlags.clean); err != nil {
+		// Link the +20 bundles to the +10 so we are building the update with the same
+		// underlying content. The only things that might change are the manifests
+		// (potentially the pack and full-file formats as well, though this is very
+		// rare).
+		ver, err = strconv.Atoi(b.MixVer)
+		if err != nil {
 			fail(err)
 		}
+		ver -= 10
+		source := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer)
+		dest := filepath.Join(b.Config.Builder.ServerStateDir, "image", strconv.Itoa(ver))
+		fmt.Println(" Copying +20 bundles to +10 bundles")
+		if err = helpers.RunCommandSilent("cp", "-al", source, dest); err != nil {
+			failf("Failed to copy +10 bundles to +20: %s\n", err)
+		}
+
+		// Set the format back to old for the actual build update
+		b.State.Mix.Format = oldFormat
 
+		// Update mixer to build version +10, the version we want to output for old format
 		ver, err = strconv.Atoi(b.MixVer)
 		if err != nil {
 			fail(err)
 		}
-
-		// Build the +20 update so we don't have to switch tooling in between
+		if err = b.UpdateMixVer(ver - 10); err != nil {
```

Comment:
Even if you do want to keep the hardcoded +-10 increments/decrements, at least save the versions to their own variables so I don't have to keep figuring out where `ver` is at with all the `ver -= 10` and `ver - 10`....

Reply:
hehe done sorry